source/extensions/filters/http/grpc_http1_reverse_bridge/filter.cc

```diff
@@ -0,0 +1,174 @@
+#include "extensions/filters/http/grpc_http1_reverse_bridge/filter.h"
+
+#include "common/common/enum_to_int.h"
+#include "common/grpc/codec.h"
+#include "common/grpc/common.h"
+#include "common/grpc/status.h"
+#include "common/http/headers.h"
+#include "common/http/utility.h"
+
+#include "extensions/filters/http/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcHttp1ReverseBridge {
+
+namespace {
+Grpc::Status::GrpcStatus grpcStatusFromHeaders(Http::HeaderMap& headers) {
+  const auto http_response_status = Http::Utility::getResponseStatus(headers);
+
+  // Notably, we treat an upstream 200 as a successful response. This differs
+  // from the standard but is key in being able to transform a successful
+  // upstream HTTP response into a gRPC response.
+  if (http_response_status == 200) {
+    return Grpc::Status::GrpcStatus::Ok;
+  } else {
+    return Grpc::Utility::httpToGrpcStatus(http_response_status);
+  }
+}
+
+void adjustContentLength(Http::HeaderMap& headers,
+                         std::function<uint64_t(uint64_t value)> adjustment) {
+  auto length_header = headers.ContentLength();
+  if (length_header != nullptr) {
+    uint64_t length;
+    if (StringUtil::atoul(length_header->value().c_str(), length)) {
+      length_header->value(adjustment(length));
+    }
+  }
+}
+} // namespace
+
+Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool end_stream) {
+  // Short circuit if header only.
+  if (end_stream) {
+    return Http::FilterHeadersStatus::Continue;
```

Comment:
Misses two line of coverage here and below. Mind add test for them?
https://138185-65214191-gh.circle-artifacts.com/0/build/envoy/generated/coverage/coverage.source_extensions_filters_http_grpc_http1_reverse_bridge_filter.cc.html